frontend/src/container/FormAlertRules/index.tsx

```diff
@@ -169,6 +212,31 @@ function FormAlertRules({
 		return retval;
 	}, [t, promQueries, queryCategory]);
 
+	const validateChQueryParams = useCallback((): boolean => {
```

Comment:
can we move this one in util ?

Reply:
Let me think over it and get back. I do understand moving it out reduces the complexity of the form.  

My thought process when definining utils was - to mainly keep a space for handling data transformations between NewWidget (Dashbaord) and Alert builder format. though usually I prefer not to use  utils as it provides no context.  (may be I could rename the utils.ts to transform.ts )

This block is in a sense validation of input form. Or you may call it validation of IBuilderQuery object itself. so i could move it to src/types/api/alerts where IBuilderQuery is defined but I did feel comfortable keeping it in the current place to use form elements easily.

Reply:
So i thought about this, I am not seeing any special value in moving the form specific logic out of the component. If you are feeling strongly about this, lets discuss.  I can probably do this after we push this PR through, as this may not be critical.